cnf-tests/testsuites/pkg/machineconfigpool/machineconfigpool.go

```diff
@@ -25,17 +25,18 @@ func WaitForCondition(
 	conditionStatus corev1.ConditionStatus,
 	timeout time.Duration,
 ) error {
+	klog.Infof("Waiting for MCP %s: %s == %s", mcp.Name, conditionType, conditionStatus)
 	return wait.PollImmediate(3*time.Second, timeout, func() (bool, error) {
 		mcpUpdated, err := cs.MachineConfigPools().Get(context.Background(), mcp.Name, metav1.GetOptions{})
```

Comment:
I was wondering why we call the mcp var here mcpUpdated,
because we might want to check other condition other than updated.
Also, why do we do "Get" if we already have the object as an input?

Reply:
Both questions have the same answer: we need to fetch a "fresh" copy of the object every time the loop runs. Otherwise, the mcp.Status.Conditions field would never change. I guess `mcpUpdated` has been used to indicate it's an "updated version of the mcp". That variable name comes from a previous commit, not this change.

I can the name of that variable as part of this PR if you have proposals

Reply:
thanks for the clarification. the name change is no must.